Scripts/Runtime/Entities/TaskDriver/TaskSet/Job/JobType/TaskCancelJobForDeferExtension.cs

```diff
@@ -93,61 +109,36 @@ private static void ValidateReflectionData(IntPtr reflectionData)
         }
 
         //*************************************************************************************************************
-        // WRAPPER STRUCT
+        // PRODUCER
         //*************************************************************************************************************
 
-        internal struct WrapperJobStruct<TJob, TInstance>
-            where TJob : struct, ITaskCancelJobForDefer<TInstance>
-            where TInstance : unmanaged, IEntityKeyedTask
+        internal struct WrapperJobProducer<TJob>
+            where TJob : struct, ITaskCancelJobForDefer
         {
-            private const int UNSET_NATIVE_THREAD_INDEX = -1;
-
-            internal TJob JobData;
-            internal DataStreamCancellationUpdater<TInstance> CancellationUpdater;
-            [NativeSetThreadIndex] internal readonly int NativeThreadIndex;
+            // ReSharper disable once StaticMemberInGenericType
+            internal static readonly SharedStatic<IntPtr> JOB_REFLECTION_DATA = SharedStatic<IntPtr>.GetOrCreate<WrapperJobProducer<TJob>>();
 
-            public WrapperJobStruct(ref TJob jobData, CancelScheduleInfo<TInstance> scheduleInfo)
+            [BurstDiscard]
+            internal static void Initialize()
             {
-                JobData = jobData;
-                CancellationUpdater = scheduleInfo.CancellationUpdater;
-                NativeThreadIndex = UNSET_NATIVE_THREAD_INDEX;
+                if (JOB_REFLECTION_DATA.Data == IntPtr.Zero)
+                {
+                    JOB_REFLECTION_DATA.Data = JobsUtility.CreateJobReflectionData(
+                        typeof(WrapperJobProducer<TJob>),
+                        typeof(TJob),
+                        (ExecuteJobFunction)Execute);
+                }
```

Comment:
How do calls to this work if it's being burst discarded?
It seems to generate state that is assumed to exist in other parts of the code.

Are you sure this works when burst isn't a JIT operation and AOT generated during a non-editor build?

Reply:
It's not being called from Burst. It happens as part of the `EarlyJobInit` code gen post processor that Unity runs. On startup it will get initialized from the main thread. 

All the existing Unity jobs like `IJobParallelForDefer` etc have `BurstDiscard` on their `Initialize` functions, so we're just "when in Rome..."ing it in ours.